src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1533,3 +1533,37 @@ def list_consumption_locations():
     client = web_client_factory()
     regions = client.list_geo_regions(sku='Dynamic')
     return [{'name': x.name.lower().replace(" ", "")} for x in regions]
+
+
+def enable_zip_deploy(resource_group_name, name, src, slot=None):
+    # if src is not of .zip extension throw exception
+    if not src.endswith('.zip'):
+        raise CLIError("src '{}' path needs to be .zip file".format(src))
+
+    client = web_client_factory()
+    user_name, password = _get_site_credential(client, resource_group_name, name)
+    scm_url = _get_scm_url(resource_group_name, name, slot)
+    zip_url = scm_url + '/api/zipdeploy'
+
+    import urllib3
+    headers = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
+    headers['content-type'] = 'application/octet-stream'
+
+    import requests
+    import os
+    # Read file content
+    fs = open(os.path.realpath(os.path.expanduser(src)), 'rb')
+    zip_content = fs.read()
+
+    try:
+        r = requests.post(zip_url, data=zip_content, headers=headers)
+        if r.status_code != 200:
+            raise CLIError("Zip deployment {} failed with status code '{}' and reason '{}'".format(
+                zip_url, r.status_code, r.text))
+    finally:
+        fs.close()
+
+    # on successful deployment navigate to the app
+    logger.warning('Zip deployment successful')
+    logger.warning('Launching app ...')
+    return view_in_browser(resource_group_name, name, slot)
```

Comment:
I suggest we expose a flag to control whether to launch the browser; as launching a UI won't work well with scripting. I suggest default the option to `False`

Reply:
i changed this to not launch the browser but show the latest deployment result{
    "id": "8fe928b7c9554df492c996c198b510ad",
    "status": 4,
    "status_text": "",
    "author_email": "N/A",
    "author": "N/A",
    "deployer": "Zip-Push",
    "message": "Created via zip push deployment",
    "progress": "",
    "received_time": "2017-11-01T21:10:31.7418815Z",
    "start_time": "2017-11-01T21:10:31.9449957Z",
    "end_time": "2017-11-01T21:10:33.0718276Z",
    "last_success_end_time": "2017-11-01T21:10:33.0718276Z",
    "complete": true,
    "active": true,
    "is_temp": false,
    "is_readonly": true,
    "url": "[https://sisirap-zipdeploymenttest.scm.azurewebsites.net/api/deployments/latest](https://sisirap-zipdeploymenttest.scm.azurewebsites.net/api/deployments/latest)",
    "log_url": "[https://sisirap-zipdeploymenttest.scm.azurewebsites.net/api/deployments/latest/log](https://sisirap-zipdeploymenttest.scm.azurewebsites.net/api/deployments/latest/log)",
    "site_name": "sisirap-ZipDeploymentTest"
} 

---
In reply to: [148144171](https://github.com/Azure/azure-cli/pull/4700#discussion_r148144171) [](ancestors = 148144171)